Forge/src/main/java/me/ferdz/placeableitems/init/PlaceableItemsTileEntityTypeRegistry.java

```diff
@@ -0,0 +1,47 @@
+package me.ferdz.placeableitems.init;
+
+import java.util.Collection;
+import java.util.function.Supplier;
+
+import com.google.common.collect.Multimap;
+import com.google.common.collect.MultimapBuilder;
+
+import me.ferdz.placeableitems.PlaceableItems;
+import me.ferdz.placeableitems.tileentity.FluidHolderTileEntity;
+import me.ferdz.placeableitems.tileentity.StackHolderTileEntity;
+
+import net.minecraft.block.Block;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+
+@EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
+public final class PlaceableItemsTileEntityTypeRegistry {
+
+    private static final Multimap<Class<? extends TileEntity>, Block> VALID_BLOCKS = MultimapBuilder.hashKeys().arrayListValues().build();
+
+    private PlaceableItemsTileEntityTypeRegistry() { }
+
+    @SubscribeEvent
+    public static void registerTE(RegistryEvent.Register<TileEntityType<?>> e) {
+        e.getRegistry().registerAll(
+                registerWithBlocks(StackHolderTileEntity::new, StackHolderTileEntity.class).setRegistryName(PlaceableItems.MODID, "writable_book_block"),
```

Comment:
Looking back on it, I don't know why this is called writable_book_block. `StackHolder` applies to many other blocks, such as the swords. Shouldn't this be renamed to something a bit more appropriate?

Reply:
It should. You're correct. The issue being that, without a data fixer, it will break any existing tile entities of this type. For instance, any swords (potentially with lots of super special enchantments!), when broken, will not drop any item. Because data fixers are so poorly documented - in fact, not at all documented - it makes it very difficult to make one in order to convert the tile entity IDs.

Reply:
I think we can do a release with a warning for users to remove the swords from the world before updating themselves. Either on a minor 1.14 or on the major 1.15.
Okay then, let's keep a TODO on it and I'll think about doing it on a minor or major update